crates/rpc/rpc-types/src/eth/filter.rs

```diff
@@ -245,18 +247,29 @@ impl FilterBlockOption {
 }
 
 /// Filter for
-#[derive(Default, Debug, PartialEq, Eq, Clone)]
+#[derive(Default, Debug, Clone)]
 pub struct Filter {
     /// Filter block options, specifying on which blocks the filter should
     /// match.
     // https://eips.ethereum.org/EIPS/eip-234
     pub block_option: FilterBlockOption,
     /// Address
     pub address: FilterSet<Address>,
-    /// Topics (maxmimum of 4)
+    /// Topics (maximum of 4)
     pub topics: [Topic; 4],
+    /// Reorged logs
+    pub reorged_logs: Arc<Mutex<Option<Vec<Log>>>>,
 }
```

Comment:
this is the rpc type, this type should remain unchanged, there's no regorged logs field in the rpc object

this should be moved to the filter implementation